### PR TITLE
Refactor(handler part 3):  Feat/handler init service mixin v2

### PR DESCRIPTION
**Pull request by 1larity**
## Summary

This PR continues functional decomposition of `acestep/handler.py` by extracting init/offload-adjacent logic into `InitServiceMixin`, while preserving `AceStepHandler` API and runtime behavior.

It also includes review-driven hardening fixes and targeted unit test coverage expansion.

## Core Work

- Extracted init-service utility methods from `acestep/handler.py` to:
  - `acestep/core/generation/handler/init_service.py`
- Kept class API surface stable via mixin composition:
  - `AceStepHandler(InitServiceMixin, LoraManagerMixin, ProgressMixin)`
- Updated mixin exports:
  - `acestep/core/generation/handler/__init__.py`

Moved methods include:
- device/cache/sync helpers
- quantized parameter helpers
- recursive model/device move utilities
- model load/offload context manager

## Follow-up Fixes (Review-Driven)

- Corrected `get_available_checkpoints` docstring to reflect actual return type/behavior.
- Added XPU support in cache clearing path (`_empty_cache`).
- Preserved `torch.nn.Parameter` registration in recursive parameter move paths.
- Fixed quantized fallback path to return `torch.nn.Parameter` (not bare tensor).
- Hardened malformed device fallback in `_is_on_target_device` (warn + conservative `False`).

## Tests

Added/expanded focused unit tests in:
- `acestep/core/generation/handler/init_service_test.py`

Coverage includes:
- device type normalization and alias behavior
- CUDA/MPS/XPU cache and synchronize routing
- flash-attention availability checks
- malformed device fallback behavior
- parameter type preservation in recursive moves
- quantized fallback preserving `Parameter`

## Validation

Passed locally:
- `python -m unittest acestep/core/generation/handler/init_service_test.py`
- `python -m unittest acestep/core/generation/handler/init_service_test.py acestep/core/lora/service_test.py acestep/core/generation/handler/lora_integration_test.py`

UI verification:
- Manual UI checks passed for impacted handler flows (including initialization/training interaction paths exercised during this slice).

## Scope Notes

- This is a narrow FD slice; no feature additions intended.
- No intended behavioral changes beyond explicit hardening fixes above.
- Branch rebased onto latest `upstream/main` before push.

## Summary by CodeRabbit

* **New Features**
  * Improved initialization and device management in the handler, enhancing model loading, device placement and offloading across accelerators.

* **Bug Fixes / Behavior Changes**
  * Consolidated initialization flow and simplified device/offload handling; some previously available utility behaviors were relocated or removed, which may alter edge-case behavior.

* **Tests**
  * Added extensive tests validating device handling, checkpoint discovery, model detection and offload scenarios.

Follow-up note: this PR intentionally normalizes _device_type() to backend tokens (cuda/mps/xpu/cpu) for cache/sync/memory routing correctness. Full indexed device refs (e.g., cuda:0, cuda:1) are still used in placement/comparison paths. However, external indexed-device handling is not yet centralized and should be addressed in a follow-up (separate helper returning both backend token + full device ref, plus multi-GPU indexed tests) to prevent future regressions around external refs.

### CODE REVIEW

**Comment by 1larity** [comment]
Maintainer note: the “force-pushed feat/handler-init-service-mixin from 886b2c2 to 3303b1c” event is expected and not a scope regression. I rebased this work onto current upstream/main, which rewrote commit SHAs but did not expand scope. Current effective diff is limited to the init-service decomposition slice only (4 files): acestep/core/generation/handler/__init__.py, acestep/core/generation/handler/init_service.py, acestep/core/generation/handler/init_service_test.py, and acestep/handler.py. No unrelated files/features are included.

**Review by ChuxiJ — APPROVED** [review]
